content/en/blog/2023/ambient-performance-savings/index.md

```diff
@@ -0,0 +1,133 @@
+
+- [Ztunnels](/blog/2023/rust-based-ztunnel/) are a brand new Istio component written in Rust that are designed to be fast, secure, and lightweight. Ztunnels are deployed per node on a cluster and enable the most basic service mesh configurations for L4 features such as mTLS, telemetry, authentication, and L4 authorizations.
+- [Waypoint proxies](/blog/2023/waypoint-proxy-made-simple/) provide L7 mesh features such as VirtualService routing, L7 telemetry, and L7 authorizations policies. Waypoints are still based on Envoy and are deployed at the namespace level per ServiceAccount.
+
+These ztunnels and waypoint proxies work in tandem to replace sidecars in the Istio service mesh. So let’s take a closer look at how the two architectures compare in the tests above.
+
+## A Closer Look
+
+{{< image width="100%"
+    link="cpu-by-pod.png"
+    caption="CPU usage by pod for sidecar and ambient pods"
+    >}}
+
+{{< image width="100%"
+    link="memory-by-pod.png"
+    caption="Memory usage by pod for sidecar and ambient pods"
+    >}}
+
+{{< image width="100%"
+    link="resource-usage.png"
+    caption="CPU and Memory usage for cluster"
+    >}}
+
+Let’s start by looking at CPU usage by pod. In the sidecar scenarios, the containers utilizing the most CPU resources are the [frontend](https://github.com/GoogleCloudPlatform/microservices-demo/tree/main/src/frontend) service sidecars, which is main server hit by the load generators and also responsible for making any subsequent calls to other underlying microservices.
+
+In the ambient scenarios, a particular ztunnel's performance depends on the distribution of services across the nodes in the cluster. In the ambient with both ztunnel and waypoint proxy scenario, there is an increase in average CPU usage per ztunnel as traffic is now directed to waypoint proxies. The waypoint proxies themselves consume more resources than their sidecar counterparts since there are only three instances processing all traffic in the namespace versus one per pod in the sidecar scenario.
+
+Next is memory usage by pod. In all Istio scenarios, memory usage stays relatively constant for each pod during the test runs. In both ambient scenarios, the L4 ztunnel consumes less memory than a sidecar. In fact, in the L4 only scenario, almost all three ztunnel instances consume less memory than a single sidecar. Waypoint proxies consume a similar amount of resources as sidecars do, but once again see a minor increase due to the same reasons outlined above for CPU. So what do these per pod usages mean for the totals across the cluster? Everything.
+
+{{< image width="100%"
+    link="total-cpu.png"
+    caption="Total CPU usage for sidecar and ambient scenarios"
+    >}}
+
+{{< image width="100%"
+    link="total-memory.png"
+    caption="Total memory usage for sidecar and ambient scenarios"
```

Comment:
I found this confusing too - @GregHanson can you clarify?

Reply:
Thinking loud, if this is interpreted correctly, could the difference cause by the fact the load generator runs in the app?

Reply:
Or in ambient L4/L4+L7, the total ram of workloads don't include the app pods themselves, only ztunnels and waypoints?

Reply:
this same problem was present in the fortio w/ httpbin example too - so it is not related to loadgenerator service

Reply:
the query for the green `Total of XXX of workloads` is very straightforward:
```
sum(container_memory_working_set_bytes{namespace=~"test.*"})
```
I have another test running now to compare numbers, but it almost seems like the memory for sidecar containers are somehow getting counted twice in the metric?

Reply:
![Screenshot 2023-07-11 at 11 04 23 AM](https://github.com/istio/istio.io/assets/19473391/4e8ccecf-2826-4bd9-8ceb-7f45517c5780)
@howardjohn I added a query to keep an eye on just the application container memory usage:
```
sum(container_memory_working_set_bytes{namespace=~"test.*", container!~"istio-proxy"}) by (pod)
```
and the application containers are just using more memory with sidecars.  And since the memory increase is present in my original fortio/httpbin example - I think I can say the increase is client/server independent

any thoughts behind the increase?

Reply:
discussed offline - there was a bug in how application totals were being queried resulting in the men and CPU metrics are being calculated, resulting in those metrics being counted twice.
```
sum(container_memory_working_set_bytes{namespace=~"test.*"})
```
vs
```
sum(container_memory_working_set_bytes{namespace=~"test.*", container!=""})
```

NOTE: the _application_ CPU/MEM totals were the only chart with this error, and these values were not used in any of the above calculations

Updating charts shortly

Reply:
charts updated

Reply:
Thanks, so this concern is addressed? @howardjohn can you confirm it from your perspective?